recipes/openvdb/8.1.0/conanfile.py

```diff
@@ -0,0 +1,205 @@
+import os
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.33.0"
+
+
+class OpenVDBConan(ConanFile):
+    name = "openvdb"
+    description = (
+        "OpenVDB is an open source C++ library comprising a novel hierarchical data "
+        "structure and a large suite of tools for the efficient storage and "
+        "manipulation of sparse volumetric data discretized on three-dimensional grids."
+    )
+    license = "MPL-2.0"
+    topics = ("voxel", "voxelizer", "volume-rendering", "fx")
+    homepage = "https://github.com/AcademySoftwareFoundation/openvdb"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_blosc": [True, False],
+        "with_zlib": [True, False],
+        "with_log4cplus": [True, False],
+        "simd": [None, "sse42", "avx"],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_blosc": True,
+        "with_zlib": True,
+        "with_log4cplus": False,
+        "simd": None,
+    }
+
+    _cmake = None
+
+    _compilers_min_version = {
+        "msvc": "19.10",
+        "Visual Studio": "15",  # Should we check toolset?
+        "gcc": "6.3.1",
+        "clang": "3.8",
+        "apple-clang": "3.8",
+        "intel": "17",
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def _check_compilier_version(self):
+        compiler = str(self.settings.compiler)
+        version = tools.Version(self.settings.compiler.version)
+        if version < self._compilers_min_version[compiler]:
+            raise ConanInvalidConfiguration("%s requires a %s version greater than %s" % (self.name, compiler, self._compilers_min_version[compiler]))
```

Comment:
We like to print a warning for the user when the compiler is unknown... conan-io/conan#8002 is our usual copy-paste